### PR TITLE
GUACAMOLE-773: Update Java and JavaScript dependencies to latest stable and compatible versions.

### PR DESCRIPTION
**Pull request by mike-jumper**
This change redoes the update that was performed previously, updating yet further to the latest versions currently available so long as those versions are stable and compatible.

Beware that this will likely end up conflicting when merged back to `master` due to the addition of Jackson's YAML support within its `dep-coordinates.txt`. That will need to be updated to reflect both changes, ie:

```
com.fasterxml.jackson.core:jackson-databind:jar:2.13.1
com.fasterxml.jackson.core:jackson-core:jar:2.13.1
com.fasterxml.jackson.core:jackson-annotations:jar:2.13.1
com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.13.1
com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.13.1
```

Once _that_ happens, builds of `master` will fail due to the licensing documentation for SnakeYAML being for version 1.27, whereas the newer YAML support for Jackson uses 1.28. As it's the multi-LDAP support that touches that, I think that will have to be added via a separate PR against specifically `master`. I've preemptively created a PR for that, as well, though currently marked as draft until needed (see #670).

### CODE REVIEW

File: extensions/guacamole-auth-ldap/pom.xml [hunk 1]
```diff
@@ -51,7 +51,21 @@
...
+            <version>2.1.0</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.apache.mina</groupId>
+                    <artifactId>mina-core</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <!-- Override version of Apache Mina specified by Apache Director API
```
Comment (necouchman) [inline extensions/guacamole-auth-ldap/pom.xml:63]: *Directory

**Review by necouchman — CHANGES_REQUESTED** [review]
Only saw one minor issue. Also going to merge it locally and make sure the build works as expected.

File: extensions/guacamole-auth-ldap/pom.xml (hunk 1 above)
Comment (mike-jumper) [reply extensions/guacamole-auth-ldap/pom.xml]: Fixed via rebase.

**Comment by necouchman** [comment]
RADIUS module seems to have a license issue:

```
[INFO] --- exec-maven-plugin:3.0.0:exec (generate-license-files) @ guacamole-auth-radius ---
Processing runtime dependencies to produce LICENSE and NOTICE. Output will be within "/home/guacbuild/guacamole-client/extensions/guacamole-auth-radius/target/licenses".
ERROR: License information missing for commons-logging:commons-logging:jar:1.2
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:982)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:929)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:457)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

**Comment by mike-jumper** [comment]
> RADIUS module seems to have a license issue: ...

Ah - I bet I need to un-remove the information for some of the other libraries that ended up being upgraded otherwise. Too bad JRadius is so out-of-date...

**Comment by mike-jumper** [comment]
OK - I've fixed the RADIUS issue via rebase.

**Review by necouchman — APPROVED** [review]

**Comment by necouchman** [comment]
Okay, this has been merged, and I've pushed to master with the resolved conflict. Feel free to convert that un-draft #670 whenever you like.

**Comment by necouchman** [comment]
Ah, although I may have missed the addition of the YAML support into the Jackson dependency, so you may have to open another pull to correct that in the `master` branch. Sorry about that, missed that one....

**Comment by mike-jumper** [comment]
> Ah, although I may have missed the addition of the YAML support into the Jackson dependency, so you may have to open another pull to correct that in the `master` branch. Sorry about that, missed that one....

No worries - that and the SnakeYAML update should be ready on the other PR now, as well.

**Pull request merged**